examples/some-verifier/src/db.rs

```diff
@@ -242,13 +253,45 @@ impl Database {
         let mut client = self.client.write().await;
         let transaction = client.transaction().await?;
 
-        let statement = format!(
+        // Clear pre-existing verifications with overlapping credentials;
+        let (usings, wheres, cred_ids) = Platform::SUPPORTED_PLATFORMS
+            .into_iter()
+            .filter_map(|p| entry.platform_entry(p).map(|e| (p, e)))
+            .enumerate()
+            .fold(
+                (String::new(), String::new(), Vec::new()),
+                |(mut usings, mut wheres, mut cred_ids), (i, (platform, entry))| {
+                    write!(usings, ", {}", platform.table_name()).expect("can write to String");
+                    write!(
+                        wheres,
+                        " OR ({0}.{VERIFICATION_ID_COLUMN}={VERIFICATIONS_TABLE}.{ID_COLUMN} AND {0}.{CRED_ID_COLUMN}=${1})",
+                        platform.table_name(),
+                        i + 1
+                    )
+                    .expect("can write to String");
+                    cred_ids.push(entry.cred_id.public_key.as_bytes() as &(dyn ToSql + Sync));
+                    (usings, wheres, cred_ids)
+                },
+            );
+
+        // Remove leading ','
+        let usings = usings.split_once(',').map(|(_, res)| res).unwrap_or("");
+        // Remove leading "OR"
+        let wheres = wheres.split_once(" OR").map(|(_, res)| res).unwrap_or("");
+
+        let delete_statement =
+            format!("DELETE FROM {VERIFICATIONS_TABLE} USING{usings} WHERE{wheres}");
```

Comment:
Not that it matters much, but I think its more readable to do 
```suggestion
        let usings = usings.split_once(', ').map(|(_, res)| res).unwrap_or("");
        // Remove leading "OR"
        let wheres = wheres.split_once(" OR ").map(|(_, res)| res).unwrap_or("");

        let delete_statement =
            format!("DELETE FROM {VERIFICATIONS_TABLE} USING {usings} WHERE {wheres}");
```

As it's easy to see that there is whitespace between `USING` and the string inserted next to it.

Reply:
I revised this a bit to avoid the awkward "remove first OR" thing. Need to test.

Reply:
Looks good and is more readable, in my opinion. Maybe I should cool it on the cool iterator features myself 😄